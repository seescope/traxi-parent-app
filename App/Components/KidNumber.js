import React, { PropTypes } from 'react';
import { Text, StyleSheet } from 'react-native';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import { isIOS } from '../Utils';

const KidNumberStyle = StyleSheet.create({
  text: {
    color: WHITE,
    textAlign: 'center',
    fontWeight: '300',
    fontSize: isIOS ? 17 : 12,
    backgroundColor: TRANSPARENT,
  },
});

const formatNumber = (number) =>
    `0${number.slice(3, 6)} ${number.slice(6, 9)} ${number.slice(9, 12)}`;

const KidNumber = ({ children }) => (
  <Text style={KidNumberStyle.text}>{formatNumber(children)}</Text>
);

KidNumber.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]).isRequired,
};

export default KidNumber;
