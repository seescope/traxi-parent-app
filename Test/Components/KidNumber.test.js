import React from 'react';
import renderer from 'react-test-renderer';
import KidNumber from '../../App/Components/KidNumber';

describe('<KidNumber>', () => {
  it('renders a kid\'s number', () => {
    const TEST_NUMBER = '+61412345678';
    const tree = renderer.create(<KidNumber>{TEST_NUMBER}</KidNumber>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
