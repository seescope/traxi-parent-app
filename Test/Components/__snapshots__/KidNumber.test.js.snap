exports[`<KidNumber> renders a kid's number 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#FFF",
      "fontFamily": "Raleway-Light",
      "fontSize": 17,
      "textAlign": "center",
    }
  }>
  +61 412 345 678
</Text>
`;
